packages/openactive-integration-tests/test/features/leasing/anonymous-leasing/implemented/lease-opportunity-capacity-update-test.js

```diff
@@ -0,0 +1,207 @@
+    it('should return 409', () => {
+      stage.expectResponseReceived();
+      chakram.expect(responseAccessor()).to.have.status(409);
+    });
+  }
+
+  /**
+   * @param {RequestState} state
+   * @param {Number} count
+   */
+  function setOrderItemsOnState(state, count) {
+    const orderItems = parentState.orderItems.filter(oi => !oi['test:control']);
+
+    /* eslint-disable no-param-reassign */
+    state.orderItems = [];
+    let i = 0;
+
+    for (const referenceOrderItem of orderItems) {
+      for (let _ = 0; _ < count; _ += 1) {
+        const orderItem = { ...referenceOrderItem };
+        orderItem.position = i;
+        state.orderItems.push(orderItem);
+
+        i += 1;
+      }
+    }
+  }
+
+  beforeAll(async () => {
+    await parentState.fetchOpportunities(orderItemCriteria);
+    return chakram.wait();
+  });
+
+  describe('Get Opportunity Feed Items', () => {
+    (new GetMatch({
+      state: parentState, flow: parentFlow, logger, orderItemCriteria,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+  });
+
+  describe('Lease three items anonymously (succeed)', () => {
+    const state = new RequestState(logger);
+    const flow = new FlowHelper(state, {
+      stagesToSkip: new Set(['getMatch']),
+    });
+
+    beforeAll(() => {
+      setOrderItemsOnState(state, 3);
+    });
+
+    describe('C1', () => {
+      const c1 = (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .successChecks()
+        .validationTests();
+
+      // it should not take into account leased opportunities on this order
+      itShouldHaveCapacity(5, c1, () => state.c1Response);
+    });
+  });
+
+  describe('Lease ten items (fail)', () => {
+    const state = new RequestState(logger);
+    const flow = new FlowHelper(state, {
+      stagesToSkip: new Set(['getMatch']),
+    });
+
+    beforeAll(() => {
+      setOrderItemsOnState(state, 10);
+    });
+
+    describe('C1', () => {
+      const c1 = (new C1({
+        state, flow, logger,
+      }))
+        .beforeSetup()
+        .validationTests();
+
+      itShouldHaveCapacity(2, c1, () => state.c1Response);
+      itShouldReturn409Conflict(c1, () => state.c1Response);
+
+      it('should return correct numbers of OpportunityCapacityIsReservedByLeaseError and OpportunityHasInsufficientCapacityError', () => {
+        c1.expectResponseReceived();
+
+        const errors = state.c1Response.body.orderedItem.map(oi => oi.error && oi.error[0] && oi.error[0]['@type']);
+        const factor = errors.length / 10;
+
+        const count = (array, value) => array.filter(x => x === value).length;
+        chakram.expect(count(errors, undefined)).to.equal(factor * 2);
+        chakram.expect(count(errors, 'OpportunityCapacityIsReservedByLeaseError')).to.equal(factor * 3);
+        chakram.expect(count(errors, 'OpportunityHasInsufficientCapacityError')).to.equal(factor * 5);
```

Comment:
```suggestion
        expect(count(errors, undefined)).to.equal(factor * 2);
        expect(count(errors, 'OpportunityCapacityIsReservedByLeaseError')).to.equal(factor * 3);
        expect(count(errors, 'OpportunityHasInsufficientCapacityError')).to.equal(factor * 5);
```

Preference for chai where possible